maze_transformer/training/training.py

```diff
@@ -38,19 +40,23 @@ class TRAIN_SAVE_FILES:
     )
 
 
+def collate_batch(batch: list[SolvedMaze], config: MazeDatasetConfig) -> list[str]:
+    # Perf could be improved by vectorizing this
```

Comment:
im not sure if making this a list comprehension (I assume this is what you mean by vectorizing given that I dont think this is numpy-vectorizable) would make this faster, but it would be cleaner

Reply:
I did mean the numpy/pytorch version, and I realize that it would be a big chunk of work... but it seems maybe possible? The connection_list and solution are already numpy arrays, so it seems possible that would could take a batch of maze objects and turn it into a batch of tokens (the integer kind) using only tensor operations. I could totally be wrong about the feasibility here. This would be TONS of work, and my benchmarking shows that this collation contributes less than 10% to the runtime of the overall training loop, so definitely not worth it right now!